src/ucp/core/ucp_request.inl

```diff
@@ -507,14 +507,21 @@ ucp_recv_desc_init(ucp_worker_h worker, void *data, size_t length,
 }
 
 static UCS_F_ALWAYS_INLINE ucp_lane_index_t
-ucp_send_request_get_next_am_bw_lane(ucp_request_t *req)
+ucp_send_request_get_next_am_bw_lane(ucp_request_t *req, int reset)
 {
     ucp_lane_index_t lane;
 
     /* at least one lane must be initialized */
     ucs_assert(ucp_ep_config(req->send.ep)->key.am_bw_lanes[0] != UCP_NULL_LANE);
 
-    lane = (req->send.tag.am_bw_index >= UCP_MAX_LANES) ?
+    if (req->send.pending_lane != UCP_NULL_LANE) {
+        /* if pending lane is not NULL - we are in pending mode,
+         * called from arbiter. do not allow change lane due
```

Comment:
As i remember, arbiter is a UCT level ubstaction, let's don't mix it with UCP

Reply:
are you what about?
we just check here is routine called from arbiter or not, in case if arbiter called routine - do not change lane

Reply:
I mean it's better do not mention UCS arbiter used from UCT in UCP...

Reply:
will be fixed later, when pending interaction will be updated

Reply:
Then plz mark this comment with `TODO:` or `FIXME:`

Reply:
updated

Reply:
10x